src/api/ddoc/views.rst

```diff
@@ -315,6 +315,1144 @@ including the update sequence of the database from which the view was
+
+Index functions
+---------------
+
+Attempting to index by using a data field that does not exist fails. To avoid this problem, use an appropriate :ref:`index_guard_clauses <api/ddoc/view>`.
+
+.. note:: 
+    Your indexing functions operate in a memory-constrained environment where the 
+    document itself forms a part of the memory that is used in that environment. 
+    Your code's stack and document must fit inside this memory. In other words, a document 
+    must be loaded in order to be indexed. Documents are limited to a maximum size of 64 MB.
+
+.. note:: 
+    Within a search index, do not index the same field name with more than one data 
+    type. If the same field name is indexed with different data types in the same search 
+    index function, you might get an error when querying the search index that says the 
+    field "was indexed without position data." For example, do not include both of these 
+    lines in the same search index function, as they index the ``myfield`` field as two 
+    different data types: a string ``"this is a string"`` and a number ``123``.
+
+.. code-block:: javascript
+
+    index("myfield", "this is a string");
+    index("myfield", 123);
+
+The function that is contained in the index field is a JavaScript function
+that is called for each document in the database.
+The function takes the document as a parameter,
+extracts some data from it,
+and then calls the function that is defined in the ``index`` field to index that data.
+
+The ``index`` function takes three parameters, where the third parameter is optional.
+
+The first parameter is the name of the field you intend to use when querying the index,
+and which is specified in the Lucene syntax portion of subsequent queries.
+An example appears in the following query:
+
+.. code-block:: javascript
+
+    query=color:red
+
+The Lucene field name ``color`` is the first parameter of the ``index`` function.
+
+The ``query`` parameter can be abbreviated to ``q``,
+so another way of writing the query is as follows:
+
+.. code-block:: javascript
+
+    q=color:red
+
+If the special value ``"default"`` is used when you define the name,
+you do not have to specify a field name at query time.
+The effect is that the query can be simplified:
+
+.. code-block:: javascript
+
+    query=red
+
+The second parameter is the data to be indexed. Keep the following information in mind when you index your data: 
+
+- This data must be only a string, number, or boolean. Other types will cause an error to be thrown by the index function call.
+- If an error is thrown when running your function, for this reason or others, the document will not be added to that search index.
+
+The third, optional, parameter is a JavaScript object with the following fields:
+
+*Index function (optional parameter)*
+
++--------------+----------------------------------------------------------------------+----------------------------------+-----------------+
```

Comment:
If you have trouble getting this table to format correctly, you might try the list-table format:

http://docutils.sourceforge.net/docs/ref/rst/directives.html#list-table

We've not used one of these before, but Sphinx (the tool we use for building the docs) claims it's supported here: http://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#table-directives

If this works out better for you, but the build breaks still because our version of Sphinx / Docutils is too old, I'll get that fixed for you.